src/Microsoft.OpenApi/Helpers/SchemaSerializerHelper.cs

```diff
@@ -13,7 +13,10 @@ namespace Microsoft.OpenApi.Helpers
 {
     internal static class SchemaSerializerHelper
     {
-        internal static void WriteAsItemsProperties(JsonSchema schema, IOpenApiWriter writer, IDictionary<string, IOpenApiExtension> extensions)
+        internal static void WriteAsItemsProperties(JsonSchema schema,
+                                                    IOpenApiWriter writer,
+                                                    IDictionary<string, IOpenApiExtension> extensions,
+                                                    OpenApiSpecVersion version)
         {
             if (writer == null)
```

Comment:
I think we should remove the condition here, and codeQL would stop complaining below

Reply:
@maggieKimani1 what I meant here was just remove the if, but keep the check if null